src/include/processor/operator/recursive_extend/path_scanner.h

```diff
@@ -8,85 +8,116 @@ namespace kuzu {
 namespace processor {
 
 /*
- * FixedLengPathScanner scans all paths of a fixed length k. This is done by starting a backward
- * traversals from only the destination nodes in the k'th frontier (assuming the first frontier has
- * index 0) over the backwards edges stored between the frontiers stored in the BFSMorsel that
- * was used to store the data related to the BFS that was computed in the RecursiveJoin operator.
- * To identify the destination nodes in the k'th frontier, we use a semi mask that marks the
- * destination nodes (or the BFSMorsel::isAllDstTarget() function which indicates that every node is
- * a possible destination).
+ * FrontierScanner scans all dst nodes from k'th frontier. To identify the
```

Comment:
I think you should reorganize some of these comments. You repeat the same point in multiple places, e.g., that some Sanner scans dst nodes from the k'th frontier. For example, think through what should be the class comments for this base class. Maybe here you should only say that this class is a base class for some other classes or something. Or here you mention what this base class does and in the extending classes just mention what extra things they do instead of repeating. So try to reorganize the comments to avoid duplication yet covering everything you want to say about these classes.